### PR TITLE
feat(ssrf-import): Slice C.2 - extended SSRF profile events

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

Extends Slice C's SSRF event pipeline (from PR #243) with 6 additional event names and threshold-split ppO2 mapping. Same "persist events, derive at read" architecture; no schema migration.

**New SSRF event names → `ProfileEventType`:**
- `bookmark` → `bookmark`
- `safety stop` → `safetyStopStart`
- `deco stop` → `decoStopStart`
- `ceiling` / `violation` → `decoViolation` (both map, both preserved per spec's flat-mapping rule)
- `ascent` → `ascentRateWarning`
- `po2` → **threshold-split**: `value >= 1.4` → `ppO2High`, `value <= 0.18` → `ppO2Low`, mid-range (unusual) → `ppO2High` default

**Four new factory constructors** on `ProfileEvent`: `decoStop`, `decoViolation`, `ppO2High`, `ppO2Low`. All default `source: EventSource.imported`.

**Import pipeline** (`_importDives` switch) extended with 7 new cases mapping parser event maps to factory calls. `bookmark`, `safetyStopStart`, `ascentRateWarning` pass explicit `source: EventSource.imported` to override their respective factory defaults (`user` / `computed` / `computed`).

**Synthetic compound fixture** `profile-events-variety.ssrf` exercises all 8 event-type mappings in one dive.

## Stacked on PR #243

**This PR targets `feat/ssrf-slice-c` (Slice C's branch) rather than `main`**, so the diff shows only Slice C.2's commits. PR #243 must merge first; this one will auto-retarget to main afterward (or needs a rebase).

## Key design decision: `po2` threshold-split

Original spec called for flat mapping — one SSRF name → one `ProfileEventType`, no value branching. During implementation, code review flagged that Subsurface uses a single `name='po2'` event for BOTH high- and low-ppO2 alarms. Flat mapping would silently tag CCR hypoxia alarms (value ≤ 0.18 bar) as `ppO2High`, which is a safety-relevant data-correctness issue for rebreather users. Spec was amended mid-slice (with user approval) to threshold-split just this one case. Other potential threshold-splits (e.g., `ascentRateCritical` at extreme ascent rates) remain deferred to future enrichment slices.

## Explicitly NOT in this PR

- UDDF parity for event types — that's Slice C.3 (open).
- Real-world fixture validation — mapping is based on Subsurface's documented event vocabulary; synthetic fixtures prove wiring, not mapping correctness against real exports.
- DC-internal alarm names (`low battery`, `heading`, `rbt`, `model version`, `workload`) — skipped with log, not persisted.
- UI rendering — persistence only.
- Variant selection beyond `po2` — `ascent` stays flat, etc.
- New `ProfileEventType` enum values — existing 18 cover the mapping.

## Test plan

- [x] `flutter test test/features/dive_log/domain/entities/profile_event_test.dart` — 15/15 pass (includes `ppO2Low` factory + new-factories group)
- [x] `flutter test test/features/universal_import/data/parsers/subsurface_xml_parser_test.dart` — 59/59 pass (includes 14 new profile-events tests: 7 new types, boundary cases at 0.18/1.4, bookmark-without-description, non-positive guard, ceiling+violation coexistence)
- [x] `flutter test test/features/dive_import/data/services/uddf_entity_importer_test.dart` — 66/66 pass (includes happy-path for all 8 types, bookmark source override, ppO2High + ppO2Low null-guard skips)
- [x] `dart analyze lib/ test/` — zero issues
- [x] `dart format --set-exit-if-changed` on all touched files — exit 0
- [x] Manual smoke test: import a real CCR `.ssrf` file with varied events, verify `DiveProfileEvents` rows have correct `eventType` + `source='imported'`
- [x] Regression: existing SSRF imports (non-event dives) still produce identical output

### CODE REVIEW

File: lib/features/universal_import/data/parsers/subsurface_xml_parser.dart [hunk 1]
```diff
@@ -702,8 +708,74 @@ class SubsurfaceXmlParser implements ImportParser {
...
         });
+      } else if (name == 'bookmark') {
+        final timestamp = _parseDurationSeconds(event.getAttribute('time'));
+        if (timestamp == null) continue;
+        final description = event.getAttribute('description');
+        events.add({
+          'eventType': 'bookmark',
+          'timestamp': timestamp,
+          'description': ?description,
+        });
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/universal_import/data/parsers/subsurface_xml_parser.dart:719]: Not a bug. `?description` and `?value` are Dart 3.3+ null-aware map/collection elements — when the expression is null, the entry is omitted; when non-null, it is included. This repo targets SDK `^3.10.0`. `dart analyze` returns zero issues on the file and the parser tests in `subsurface_xml_parser_test.dart` (e.g. `bookmark event without description` at lines around the `profile events` group) pass against this exact syntax. See [the null-aware elements language feature](https://github.com/dart-lang/language/blob/main/accepted/3.3/null-aware-elements/feature-specification.md).

File: lib/features/dive_import/data/services/uddf_entity_importer.dart [hunk 2]
```diff
@@ -1261,6 +1264,111 @@ class UddfEntityImporter {
...
+              break;
+
+            case 'ascentRateWarning':
+              events.add(
+                ProfileEvent.ascentRateWarning(
+                  id: _uuid.v4(),
+                  diveId: diveId,
+                  timestamp: timestamp,
+                  depth: 0.0,
+                  rate: value ?? 0.0,
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/dive_import/data/services/uddf_entity_importer.dart]: Fixed in 80d0f3ea394. `ascentRateWarning` now skips with a warning when `value` is null instead of persisting `rate: 0.0`, matching the `ppO2High`/`ppO2Low`/`setpointChange` null-guard pattern. Added a regression test `ascentRateWarning event with missing value is skipped at importer level` mirroring the existing ppO2 null-skip tests.

File: docs/superpowers/specs/2026-04-05-imported-profile-gap-priority-tracker.md [hunk 3]
```diff
@@ -106,7 +106,7 @@ Use the `Fixed` column as a working checkbox:
...
 | Sample `ppO2` | [x] | Medium | Real SSRF `po2` now maps directly into sample `ppO2` |
-| Profile events / markers | [ ] | Medium | `setpointChange` events are persisted via `DiveProfileEvents` (Slice C). Bookmarks, alarms, ceiling violations, and other SSRF event types remain unmapped. A future slice should extend `_parseProfileEvents` to cover these |
+| Profile events / markers | [ ] | Medium | `setpointChange`, `bookmark`, `safetyStopStart`, `decoStopStart`, `decoViolation` (from `ceiling` / `violation`), `ascentRateWarning` (from `ascent`), `ppO2High` and `ppO2Low` (from `po2`, threshold-split at 1.4 / 0.18 bar) are now persisted via `DiveProfileEvents` (Slice C + C.2). DC-internal alarms (`low battery`, `heading`, `rbt`) remain intentionally unmapped. UDDF parity is Slice C.3. |
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply docs/superpowers/specs/2026-04-05-imported-profile-gap-priority-tracker.md]: Fixed in 80d0f3ea394. Row is now prefixed with "Partially fixed:" and explicitly states why the checkbox stays unchecked (DC-internal alarms intentionally unmapped, UDDF parity pending in Slice C.3) so the tracker is unambiguous.
